client/components/domains/search-filters/tld-filter-bar.jsx

```diff
@@ -0,0 +1,194 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+import React, { Component } from 'react';
+import { includes } from 'lodash';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CompactCard from 'components/card/compact';
+import FormFieldset from 'components/forms/form-fieldset';
+import Popover from 'components/popover';
+import TokenField from 'components/token-field';
+import { recordTldFilterSelected } from './analytics';
+
+export class TldFilterBar extends Component {
+	static propTypes = {
+		availableTlds: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		filters: PropTypes.shape( {
+			tlds: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		} ),
+		lastFilters: PropTypes.shape( {
+			tlds: PropTypes.arrayOf( PropTypes.string ).isRequired,
+		} ),
+		numberOfTldsShown: PropTypes.number,
+		onChange: PropTypes.func.isRequired,
+		onReset: PropTypes.func.isRequired,
+		onSubmit: PropTypes.func.isRequired,
+		recordTldFilterSelected: PropTypes.func.isRequired,
+		showPlaceholder: PropTypes.bool.isRequired,
+	};
+
+	static defaultProps = {
+		numberOfTldsShown: 8,
+	};
+
+	state = {
+		showPopover: false,
+	};
+
+	bindButton = button => ( this.button = button );
+
+	handleButtonClick = event => {
+		const isCurrentlySelected = event.currentTarget.dataset.selected === 'true';
+		const newTld = event.currentTarget.value;
+
+		this.props.recordTldFilterSelected(
+			newTld,
+			event.currentTarget.dataset.index,
+			! isCurrentlySelected
+		);
+
+		const tldSet = new Set( [ ...this.props.filters.tlds, newTld ] );
+		if ( isCurrentlySelected ) {
+			tldSet.delete( newTld );
+		}
+
+		const tlds = [ ...tldSet ];
+		this.props.onChange( { tlds }, { shouldSubmit: true } );
+	};
+	handleFiltersReset = () => {
+		this.togglePopover();
+		this.props.onReset( 'tlds' );
+	};
+	handleFiltersSubmit = () => {
+		this.togglePopover();
+		this.props.onSubmit();
+	};
+	handleTokenChange = newTlds => {
```

Comment:
Looks like something broke here (or in the component), because I can't seem to remove the tlds anymore from the `TokenField`.

To reproduce: just add any TLD (using the buttons or token field) and try to remove them using `X`. The `Reset` button works, though.

Reply:
I can reproduce. My guess is that I must have messed up a rebase somewhere; fix incoming.